src/pspm_sf_dcm.m

```diff
@@ -128,43 +118,64 @@
 priors.muTheta(5:2:(2 * nresp + 4)) = -10;
 dim.n_theta = numel(priors.muTheta);    % nb of evolution parameters
 priors.SigmaTheta = zeros(dim.n_theta);
-for k = (4:2:(2 * nresp + 3)), priors.SigmaTheta(k, k) = 1e-2;end;
-for k = (5:2:(2 * nresp + 4)), priors.SigmaTheta(k, k) = 1e2; end;
+for k = (4:2:(2 * nresp + 3)), priors.SigmaTheta(k, k) = 1e-2;end
+for k = (5:2:(2 * nresp + 4)), priors.SigmaTheta(k, k) = 1e2; end
 priors.muPhi = phi';
 priors.SigmaPhi = zeros(dim.n_phi);
 priors.SigmaX0 = 1e-8*eye(dim.n);
 options.priors = priors;
-
-% estimate parameters
+% 4.5 estimate parameters
 c = clock;
 fprintf(['\n\nEstimating model parameters for f_SF ... \t%02.0f:%02.0f:%02.0f', ...
   '\n=========================================================\n'], c(4:6));
-[posterior, output] = VBA_NLStateSpaceModel(y(:)',u,f_fname,g_fname,dim,options);
-
-% extract parameters
-% =======================================================================
-for i=1:length(output)
-  output(i).options = rmfield(output(i).options, 'hf');
-end;
-t = posterior.muTheta(4:2:end);
-a = exp(posterior.muTheta(5:2:end) - theta(5));   % rescale
-ex = find(t < -2 | t > (numel(scr)/sr - 1)); % find SA responses the SCR peak of which is outside episode
-t(ex) = []; a(ex) = [];
-out.t = t - theta(4);                             % subtract conduction delay
-out.a = a;
-out.n = numel(find(a > threshold));
-out.f = out.n/(numel(scr)/sr);
-out.ma = mean(a(a > threshold));
-out.theta = theta;
-out.if = fresp;
-out.threshold = threshold;
-out.yhat = posterior.muX(1, :);
-out.model.posterior = posterior;
-out.model.output = output;
-out.model.u = u;
-out.model.y = y(:)';
-out.time = toc(tstart);
-% =======================================================================
+if isfield(options, 'missing')
+  ymissing = options.missing;
+else
+  ymissing = isnan(y);
+end
+ymissing_start = find(diff(ymissing)==1);
+ymissing_end = find(diff(ymissing)==-1);
+if length(ymissing_start) > length(ymissing_end)
+  ymissing_end = [ymissing_end, length(ymissing_end)];
+elseif length(ymissing_start) < length(ymissing_end)
+  ymissing_start = [1, ymissing_start];
+end
+miss_epoch = [ymissing_start(:),ymissing_end(:)];
+flag_missing_too_long = 0;
+if diff(miss_epoch, 1, 2)/sr > options.missingthresh
```

Comment:
LHS is a vector - should this be `any()`?

Reply:
In Matlab, a vector is accepted as the condition for if loop, however it is indeed more explicit to use `any()` here, therefore I have added it. Thanks for reminding me of the point. 🙂